File: cpp/PE016.cpp
#include <iostream>
using namespace std;

int main()
{
    int a[400];
    for(int i = 0; i < 400; i++)
        a[i] = -1;
    a[0] = 1;
    for(int i = 0; i < 1000; i++){
        int carry = 0;
        int j;
        for(j = 0; a[j] != -1; j++){
            int m = 2*a[j] + carry;
            a[j] = m%10;
            carry = m/10;
        }
        if(carry)
            a[j] = carry;
    }
    int sum = 0;
    for(int i = 0; a[i] != -1; i++)
        sum += a[i];
    cout << sum << endl;
    return 0;
}

File: java/PE016.java
import java.math.BigInteger;

public class PE016
{
    public static void main(String[] args)
    {
        BigInteger a = BigInteger.ONE;
        BigInteger b = new BigInteger("2");
        for(int i = 1; i <= 1000; i++) {
            a = a.multiply(b);
        }
        String c = a.toString();
        int sum = 0;
        for(int i = 0; i < c.length(); i++) {
            sum += c.charAt(i) - '0';
        }
        System.out.println(sum);
    }
}
